Usermod rotary brightness, color and color temperature

dkneisz opened the pull request:
add usermod to control brigness and color/color tempearature using a rotary encoder

- pins for rotary encoder (CLKPin and DTPin) can be defined in usermod settings
- change amount for RGB (Hue) and color temperature  can be defined in usermod settings
- button needs to be configured in LED preferences and set in usermot settings
- single press switches between brightness and color/color temperature mode
- double press switches betweeen RGB and white

blazoncek commented:
You can use core function `void colorKtoRGB(uint16_t kelvin, byte* rgb)` for conversion from K to RGB.
For other conversion functions take a look into colors.cpp

IMO it would be better to add CCT functionality to existing and very good ALT rotary encoder (I am also currently working on a few enhancements there).

dkneisz commented:
> You can use core function `void colorKtoRGB(uint16_t kelvin, byte* rgb)` for conversion from K to RGB. For other conversion functions take a look into colors.cpp
Thanks, I will look into that. This will be better than including an additional library.

> IMO it would be better to add CCT functionality to existing and very good ALT rotary encoder (I am also currently working on a few enhancements there).
I'm not sure which usermod you exactly mean, but I didn't want to change an existing usermod and decided to use an existing one as a kind of template and create a new one. That way someone can use the original if they only need the functionality provided there, if they don't need/use my enhancements. I know that there are multiple usermods for rotary encoders, and they all do different things. I chose one that (almost) fits my needs and extended it. There are some that can do more, but I don't needed it, so I didn't choose to use them. Maybe combining all functionality into one usermod and make every functionality deactivateable (is that even a word?) would be the "way to go". But then again, I don't want to change a mod that some other user made, because it may have been the intention of that user just to have this functionality. However if this functionality would be included into WLED (so not a usermod anymore), that would be a different topic.

dkneisz commented:
I changed the code to use `colorKtoRGB` instead so I don't have to include an external library. I also removed that part mentioning that in the original comment.
